Core.Tests/Commands/CommandProcessorTest.cs

```diff
@@ -0,0 +1,120 @@
+                context => throw new InvalidOperationException("this command is busted!")));
+
+            CommandResult result = await commandProcessor.Process("broken", _noArgs, MockMessage("bla"));
+
+            Assert.AreEqual("An error occurred.", result.Response);
+            var errorTextRegex = new Regex(
+                $@"^An exception occured while executing command 'broken'\. User: {Regex.Escape(_mockUser.ToString())}, Original text: bla$");
+            loggerMock.VerifyLog(LogLevel.Error, errorTextRegex, Times.Once(),
+                exception: new InvalidOperationException("this command is busted!"));
+        }
+
+        [Test]
+        public async Task TestCaseInsensitive()
+        {
+            var commandProcessor = new CommandProcessor(_nullLogger, new ArgsParser());
+            commandProcessor.InstallCommand(new CommandInfo("MiXeD", CommandUtils.StaticResponse("Hi!")));
+
+            foreach (string command in ImmutableList.Create("MiXeD", "mixed", "MIXED"))
+            {
+                CommandResult result = await commandProcessor.Process(command, _noArgs, MockMessage());
+                Assert.AreEqual("Hi!", result.Response);
+            }
+        }
+
+        [Test]
+        public async Task TestAliases()
+        {
+            var commandProcessor = new CommandProcessor(_nullLogger, new ArgsParser());
+            commandProcessor.InstallCommand(new CommandInfo("main", CommandUtils.StaticResponse("Hi!"))
+                {Aliases = new[] {"alias1", "alias2"}});
+
+            foreach (string command in ImmutableList.Create("main", "alias1", "ALIAS2"))
+            {
+                CommandResult result = await commandProcessor.Process(command, _noArgs, MockMessage());
+                Assert.AreEqual("Hi!", result.Response);
+            }
+        }
+
+        [Test]
+        public void InstallConflictName()
+        {
+            var commandProcessor = new CommandProcessor(_nullLogger, new ArgsParser());
+
+            commandProcessor.InstallCommand(new CommandInfo("a", CommandUtils.StaticResponse("Hi!")));
+            var ex = Assert.Throws<ArgumentException>(() => commandProcessor
+                .InstallCommand(new CommandInfo("A", CommandUtils.StaticResponse("Hi!"))));
+            Assert.AreEqual("The command name 'a' conflicts with: a: <no description>", ex.Message);
+        }
+
+        [Test]
+        public void InstallConflictAlias()
+        {
+            var commandProcessor = new CommandProcessor(_nullLogger, new ArgsParser());
+
+            commandProcessor.InstallCommand(new CommandInfo("a", CommandUtils.StaticResponse("Hi!"))
+                {Aliases = new[] {"x"}});
+            var ex = Assert.Throws<ArgumentException>(() => commandProcessor
+                .InstallCommand(new CommandInfo("b", CommandUtils.StaticResponse("Hi!")) {Aliases = new[] {"X"}}));
+            Assert.AreEqual("The alias 'x' conflicts with: a(x): <no description>", ex.Message);
+        }
+    }
```

Comment:
if you have an command installed as "x" and a command with an alias as "x", it should throw, worth testing for this?